eureka/S3_data_reduction/bright2flux.py

```diff
@@ -183,19 +184,30 @@ def convert_to_e(data, meta, log):
     meta:   MetaClass
         The metadata object.
     """
+    if data.shdr['BUNIT'] != 'ELECTRONS/S':
+        log.writelog('Automatically getting reference files to convert units to e/s')
+        if data.mhdr['TELESCOP'] != 'JWST':
+            raise ValueError('Currently unable to automatically download reference files for non-jwst observations!')
+        meta.photfile, meta.gainfile = download_ancil(data.filename)
+    
     if data.shdr['BUNIT'] == 'MJy/sr':
         # Convert from brightness units (MJy/sr) to flux units (uJy/pix)
         # log.writelog('Converting from brightness to flux units')
         # data = b2f.bright2flux(data, data.shdr['PIXAR_A2'])
         # Convert from brightness units (MJy/sr) to DNs
-        log.writelog('  Converting from brightness units (MJy/sr) to electrons')
-        meta.photfile = os.path.join(meta.topdir, *meta.ancildir.split(os.sep)) + '/' + data.mhdr['R_PHOTOM'][7:]
+        log.writelog('  Converting from brightness units (MJy/sr) to electrons per second (e/s)')
```

Comment:
This function is intended to return electrons, not electrons/sec.

Reply:
Huh, well HST data comes in with e/s (or so the FITS header claims) and the dn2electrons function does not multiply by the integration time, so there might be more issues here than just the log messages

Reply:
Look at lines 65-67 `hst_scan.py`.  When the files are read in, the flux is multiplied by the sample time.